streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -411,7 +411,10 @@ boolean tryToCompleteRestoration() {
                 } else {
                     // we found a restoring task that isn't done restoring, which is evidence that
                     // not all tasks are running
+                    log.debug("Task {} has not completed restoration, will check next time", task.id());
+
                     allRunning = false;
```

Comment:
My plan is to modify the condition inside `StreamThread#runOnce` so that we can still process even if `state != RUNNING` (i.e. all tasks completed restoration) but as long as `taskManager#hasRunnableActiveTask`. Does that make sense?